pwndbg/heap/ptmalloc.py

```diff
@@ -347,42 +347,42 @@ def get_heap_boundaries(self, addr=None):
         Get the boundaries of the heap containing `addr`. Returns the brk region for
         adresses inside it or a fake Page for the containing heap for non-main arenas.
         """
+        page = pwndbg.memory.Page(0, 0, 0, 0)
         brk = self.get_region()
         if addr is None or brk.vaddr < addr < brk.vaddr + brk.memsz:
-            return brk
+            # Occasionally, the [heap] vm region and the actual start of the heap are
+            # different, e.g. [heap] starts at 0x61f000 but mp_.sbrk_base is 0x620000.
+            # Return an adjusted Page object if this is the case.
+            sbrk_base = int(self.mp['sbrk_base'])
+            if sbrk_base != brk.vaddr:
+                page.vaddr = sbrk_base
+                page.memsz = brk.memsz - (sbrk_base - brk.vaddr)
+                return page
+            else:
+                return brk
         else:
-            page = pwndbg.memory.Page(0, 0, 0, 0)
             page.vaddr = heap_for_ptr(addr)
             heap = self.get_heap(page.vaddr)
             page.memsz = int(heap['size'])
             return page
 
 
-    def get_region(self,addr=None):
+    def get_region(self, addr=None):
         """
-        Finds the memory map used for heap by using mp_ structure's sbrk_base property
-        and falls back to using /proc/self/maps (vmmap) which can be wrong
-        when .bss is very large
```

Comment:
So now we don't have this case of fallback which could make us have wrong value?